libsweep/src/sweep.cc

```diff
@@ -49,54 +51,49 @@ void sweep_error_destruct(sweep_error_s error) {
   delete error;
 }
 
-sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) {
+sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) try {
   SWEEP_ASSERT(error);
-
   return sweep_device_construct(port, 115200, error);
+} catch (const std::exception& e) {
+  *error = sweep_error_construct(e.what());
```

Comment:
I think you're right - I'm not 100% sure about the lifetime for exceptions here, but your explanation makes sense. Fix is to copy the string into the error and let the `error_destruct` function do the proper cleanup.

Thanks for catching these sneaky issues, you're a great help here! :)